admin-dev/themes/new-theme/js/components/grid/extension/submit-bulk-action-extension.js

```diff
@@ -57,11 +57,48 @@ export default class SubmitBulkActionExtension {
   submit(event, grid) {
     const $submitBtn = $(event.currentTarget);
     const confirmMessage = $submitBtn.data('confirm-message');
+    const confirmTitle = $submitBtn.data('confirmTitle');
 
-    if (typeof confirmMessage !== "undefined" && 0 < confirmMessage.length && !confirm(confirmMessage)) {
-      return;
+    if (confirmMessage !== undefined && 0 < confirmMessage.length) {
+      if (confirmTitle !== undefined) {
+        this.showConfirmModal($submitBtn, grid, confirmMessage, confirmTitle);
+      } else if (confirm(confirmMessage)) {
+        this.postForm($submitBtn, grid);
+      }
+    } else {
+      this.postForm($submitBtn, grid);
     }
+  }
+
+  /**
+   * @param {jQuery} $submitBtn
+   * @param {Grid} grid
+   * @param {string} confirmMessage
+   * @param {string} confirmTitle
+   */
+  showConfirmModal($submitBtn, grid, confirmMessage, confirmTitle) {
+    const confirmButtonLabel = $submitBtn.data('confirmButtonLabel');
+    const confirmButtonClass = $submitBtn.data('confirmButtonClass');
+
+    const $modal = $('#' + grid.getId() + '_grid_confirm_modal');
+    $('.confirm-message', $modal).html(confirmMessage);
+    $('.modal-title', $modal).html(confirmTitle);
+    $('.modal-header', $modal).toggle(confirmTitle.length > 0);
 
+    const $confirmButton = $('.btn-confirm-submit', $modal);
+    $confirmButton.className = 'btn btn-confirm-submit btn-lg';
+    $confirmButton.addClass(confirmButtonClass);
+    $confirmButton.html(confirmButtonLabel);
+    $confirmButton.off('click').on('click', () => this.postForm($submitBtn, grid));
```

Comment:
Still thiking `.off('click')` isn't a good idea and can break all modules javascript event on the confirm button..

Reply:
It is required because the same modal can be shared between several submit actions, so we need to remove the previous callback to be sure only one will be called on confirm
If module need to use the same modal then they will have to do the same cleaning

Reply:
A new modal must be created depending on the action you do. You shouldn't share the same modal context, more if you allow people to add module on your app :disappointed_relieved: 
If a module want to add a custom action before the `this.postForm` and just after the first `'click'` event, he must reimplement everything. Sad story :(

Reply:
Mmm I think you are mixing too many features here for a use case that is much more simple in this case

This modal is simply a "pretty confirm" You can change the question, title, button style and that's it It's not a modal that any other module is supposed to listen to Just like when you use `confirm` in your code no modules will be waiting for the return

We don't want to module to integrate in this modal, since we have absolutely no JS hooks, event, or module integration available today anyway

However what this PR will allow is for module to add their SubmitBulkAction in the grid definition and be able to use a pretty modal only using a PHP hook

Reply:
Even if we don't have hooks, we are able to add custom javascript files in the BO, extend the theme, or anything else. This means you're able to add a custom `.on('click')` on the confirm button to trigger a special api request or a custom effect, a new ajax box loaded.
This is also why we are not able to change all our html ids even if we don't have custom html hooks :trollface:

Reply:
@PierreRambaud But the `.on('click')` targets `.btn-confirm-submit` selector 🤔so if a module wants to customize the behavior of one button, he can remove the class for it and he's safe, isn't it ?

Reply:
If a module wants to change the behavior of this button, the correct approach would be to change the grid definition, remove the existing button and add its custom button
Currently if we don't have a modal with the confirm that's what the module would need to do, it wouldn't be able to integrate its js in the middle of the execution

So yes of course any module could binf its actions on clicks, on change on any button, they can do whatever they want and break anything but we can't imagine all the unexpected behavior anyone could imagine or we can never change anything

Reply:
And indeed as Mathieu said, if one wants to override the default behavior he can always unbind our actions on the submit button

Reply:
This is exactly what I said. If he wants to keep the original behaviour and add a little action just before or after the event. He must reimplement what you've done, this means duplicate the `postForm` method.
@matks It's not targetting the `.btn-confirm-submit`, it targets the `'#GRID_ID_grid_confirm_modal .btn-confirm-submit` which is different.

As I said earlier, you disallow the capability to add an event when you click on the button, I keep thinking it's really bad for front developers :/
Two solutions:
1. Generate the modal, so no one can add custom event on it
2. One modal per action

Don't forget people can add javascript, and there is no hook / event in the BO for front developers, don't hand them the stick :(

Reply:
I agree with Pierre, let peoples be ables to add events on buttons, don't kick events off, it won't change anything about your behavior no ?

Peoples may wants to add a behavior to this button and not delete your one, if they wants to do this, they need to remove your event, add your own behavior again and then add their behavior, it doesn't look like it's a good practice at all

Reply:
But I don't remove anything!! Before you add just a `confirm` call, and absolutely NO way to integrate a behaviour before or after this confirm You are mixing things here, I didn't remove the possibility to add any behaviour on the bulk button
This confirm button is a new button in a new modal that no one has ever seen or has ever modified

Reply:
This modal can be seen as a blackbox modal confirm component, and has no intention of allowing internal customization so indeed I don't want external modules to change its behaviour